Script Analysis Examples/Test Data Analysis/Failure_Pareto_Result_Analysis.ipynb

```diff
@@ -0,0 +1,642 @@
+    "\n",
+    "df_dict = {\n",
+    "    'columns': pd.io.json.build_table_schema(df_pareto, index=False)['fields'],\n",
+    "    'values': df_pareto.values,\n",
+    "}\n",
+    "\n",
+    "pareto_graph = {\n",
+    "    \"type\": 'data_frame',\n",
+    "    'id': 'failure_pareto_results_graph',\n",
+    "    'data': df_dict,\n",
+    "    'config': {\n",
+    "        'title': 'Failure Pareto - Results by {}'.format(GROUP_BY),\n",
+    "        'graph': {\n",
+    "            'axis_labels': [GROUP_BY, 'Failure Count', 'Cumulative %'],\n",
+    "            'plots': [\n",
+    "                {'x': grouping, 'y': 'fail_count', 'style': 'BAR', 'GROUP_BY': [grouping]},\n",
+    "                {'x': grouping, 'y': 'cumulative', 'secondary_y': True, 'style': 'LINE'}\n",
+    "            ],\n",
+    "            'orientation': 'VERTICAL'\n",
+    "        }\n",
+    "    }\n",
+    "}\n",
+    "\n",
+    "result = [pareto_graph]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Record results with Scrapbook"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sb.glue('result', str(result))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generate plot PNG"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extract data from the pareto_graph dictionary\n",
+    "title = pareto_graph['config']['title']\n",
+    "axis_labels = pareto_graph['config']['graph']['axis_labels']\n",
+    "plots = pareto_graph['config']['graph']['plots']\n",
+    "\n",
+    "# Plot data\n",
+    "fig, ax1 = plt.subplots()\n",
+    "\n",
+    "for plot in plots:\n",
+    "    if plot['style'] == 'BAR':\n",
+    "        ax1.bar(\n",
+    "            df_pareto[plot['x']], \n",
+    "            df_pareto[plot['y']], \n",
+    "            label=plot['x'])\n",
+    "        ax1.set_ylabel(axis_labels[1])\n",
+    "        ax1.set_xlabel(axis_labels[0])\n",
+    "    elif plot['style'] == 'LINE':\n",
+    "        ax2 = ax1.twinx()\n",
+    "        ax2.plot(df_pareto[plot['x']], df_pareto[plot['y']], label=plot['x'], color='r')\n",
+    "        ax2.set_ylabel(axis_labels[2])\n",
+    "\n",
+    "# Title and legend\n",
+    "plt.title(title)\n",
+    "plt.legend()\n",
+    "ax1.tick_params(axis='x', labelrotation = 90)\n",
+    "\n",
+    "# Save as PNG file\n",
+    "plt.savefig(PLOT_FILE_NAME, bbox_inches='tight')\n",
+    "\n",
+    "# Show the plot (optional)\n",
+    "plt.show()\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Upload Plot to files service and link to products"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_post_request(url, body, headers = None):\n",
```

Comment:
We should use the python api for uploading files and not use the raw python `requests` library here.

Reply:
updated to use python api for uploading the file